backend/src/packages/journal-entries/journal-entry.service.ts

```diff
@@ -83,8 +85,19 @@ class JournalEntryService implements Service {
     return item.toObject();
   }
 
-  public delete(): ReturnType<Service['delete']> {
-    return Promise.resolve(true);
+  public async delete(payload: {
+    id: number;
+    user: UserAuthResponseDto;
+  }): ReturnType<Service['delete']> {
+    const journal = await this.find(payload.id);
+    if (journal.userId !== payload.user.id) {
+      throw new JournalError({
+        status: HTTPCode.BAD_REQUEST,
+        message: ExceptionMessage.INCORRECT_CREDENTIALS,
+      });
+    }
+
+    return Boolean(await this.journalEntryRepository.delete(payload.id));
```

Comment:
move async call to a separate variable like deletedCount

Reply:
done